src/createServer.js

```diff
@@ -1,3 +1,67 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('../src/convertToCase/convertToCase.js');
+const { errorTypes } = require('./errorTypes.js');
+
+const CASES = new Set(['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER']);
+
+function createServer() {
+  return http.createServer((req, res) => {
+    const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+
+    const errors = [];
+
+    const toCase = normalizedUrl.searchParams.get('toCase');
+
+    if (!toCase) {
+      errors.push({
+        message: errorTypes.toCaseRequired,
+      });
+    } else if (!CASES.has(toCase.toUpperCase())) {
+      errors.push({
+        message: errorTypes.toCaseNotSupported,
+      });
+    }
```

Comment:
Create a function that validates input for errors and move it to a separate file